stake/packages/frontend/components/NFTQuery.tsx

```diff
@@ -295,13 +295,21 @@ export const NFTQuery = ({
               : undefined
           }
         )
+
         const nftsPromises = nftsMappedArray.map(async (_, index) => {
           const tokenId = await HoprBoost.tokenOfOwnerByIndex(account, index)
           return await getNFTFromTokenId(HoprBoost, tokenId)
         })
+
         // We resolve both promises to make sure all NFTs are properly obtained
-        const nfts = (await Promise.all(nftsPromises)) || []
+        const allNfts = (await Promise.all(nftsPromises)) || []
         const redemeedNfts = (await Promise.all(redeemedNFTSPromises)) || []
+
+        // We filter out all blocked NFT types
+        const nfts = allNfts.filter((nft: NFT) => {
+          return !(nft.typeOfBoost in ['2'])
```

Comment:
`['2']` can be replaced with the array returned from the graph's api call (as in [TotalStakedBalance.tsx](https://github.com/hoprnet/hopr-devrel/tree/e3e05a2f220ea35c74db4156a8c998f963171b02/stake/packages/frontend/components/atoms/TotalStakedBalance.tsx))
with the following query
```
const QUERY_BLOCKEDTYPE = `{
  programs(first: 1) {
    blockedType
  }
}`
```
followed by the query
```
const { data } = await client.query(QUERY_BLOCKEDTYPE).toPromise()
const blockedNftTypes = data.programs.length > 0 ? data.programs[0].blockedType : []
```
This would make the frontend be always synced with the contract's block list.

Reply:
Thanks. Applied.